src/libcore/iter/mod.rs

```diff
@@ -643,6 +643,41 @@ impl<I> Iterator for Cycle<I> where I: Clone + Iterator {
             _ => (usize::MAX, None)
         }
     }
+
+    #[inline]
+    fn all<F>(&mut self, f: F) -> bool where F: FnMut(Self::Item) -> bool { self.orig.clone().all(f) }
```

Comment:
Using a new clone of `orig` instead of using `iter` is wrong here, since it'll short-circuit in a different place.

Reply:
Ah, that's true.